src/ios/PushPlugin.m

```diff
@@ -177,21 +199,21 @@ - (void)init:(CDVInvokedUrlCommand*)command;
         }];
     } else {
         NSLog(@"Push Plugin VoIP missing or false");
-        [[NSNotificationCenter defaultCenter]
-         addObserver:self selector:@selector(onTokenRefresh)
-         name:kFIRInstanceIDTokenRefreshNotification object:nil];
+        // [[NSNotificationCenter defaultCenter]
+        //  addObserver:self selector:@selector(onTokenRefresh)
+        //  name:kFIRInstanceIDTokenRefreshNotification object:nil];
```

Comment:
Removing this one is more concerning and needs confirmation. Is it safe to delete or does it need to be updated?

If you read the comments on the `onTokenRefresh` method, it sounds like an important use case.

```objc
//  FCM refresh token
//  Unclear how this is testable under normal circumstances
- (void)onTokenRefresh {
#if !TARGET_IPHONE_SIMULATOR
    // A rotation of the registration tokens is happening, so the app needs to request a new token.
    NSLog(@"The FCM registration token needs to be changed.");
    [self initRegistration];
#endif
}
```

As mentioned in the StackOverflow, they replaced the `name:kFIRInstanceIDTokenRefreshNotification` name with the follow two:

* FIRMessagingRegistrationTokenRefreshedNotification
* FIRInstallationIDDidChangeNotification

https://stackoverflow.com/a/65111763/1644129

If updating the name to `FIRMessagingRegistrationTokenRefreshedNotification` works without causing any errors, I think we should keep it.

I am worried about what happens if there was a rotation of the registration tokens and the app does not request a new token. It sounds like it would stop receiving push notifications.

But again, because you are trying to update the library from 6.x to 8.x (majors), it is possible it changed.